ui/src/app/pages/analysis/notebook-redirect.tsx

```diff
@@ -352,41 +378,6 @@ export const NotebookRedirect = fp.flow(withUserProfile(), withCurrentWorkspace(
       this.clearAndSetTimeout(() => this.incrementProgress(Progress.Loaded), 1000);
```

Comment:
One comment on this clearAndSetTimeout() pattern: I think this particular usage shows some issues with the pattern. Namely - to use it in this context, you have to know that the timeout is not currently pending on any of the other callsites. I think in this case it would be safer and easier to verify by inspection if you just had two separate timer state trackers:

this.pollTimer
this.redirectTimer

Also to be clear, it is not always necessary to call `clearTimeout`; it's effectively a no-op if the timeout has already fired. I realize I may not have addressed this point when replying to your original proposal. In general if you write your code such that it does not overwrite references to live timeouts (no need to worry about spent ones), then clearing them in the unmount section only is sufficient. I think this is probably the guidance we should use here, though it took me a while to verbalize it properly.

Reply:
good points.  updated to use setTimeout() directly for these two handles